NCS.DSS.EmploymentProgression/PostEmploymentProgression/Service/EmploymentProgressionPostTriggerService.cs

```diff
@@ -0,0 +1,74 @@
+﻿using DFC.GeoCoding.Standard.AzureMaps.Model;
+using NCS.DSS.EmploymentProgression.Cosmos.Provider;
+using NCS.DSS.EmploymentProgression.ServiceBus;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace NCS.DSS.EmploymentProgression.PostEmploymentProgression.Service
+{
+    public class EmploymentProgressionPostTriggerService : IEmploymentProgressionPostTriggerService
+    {
+        private readonly IDocumentDBProvider _documentDbProvider;
+        private readonly IServiceBusClient _serviceBusClient;
+
+        public EmploymentProgressionPostTriggerService(IDocumentDBProvider documentDbProvider, IServiceBusClient serviceBusClient)
+        {
+            _documentDbProvider = documentDbProvider;
+            _serviceBusClient = serviceBusClient;
+        }
+
+        public async Task<Models.EmploymentProgression> CreateEmploymentProgressionAsync(Models.EmploymentProgression employmentProgression)
+        {
+            if (employmentProgression == null)
+            {
+                return null;
+            }
+
+            employmentProgression.EmploymentProgressionId = Guid.NewGuid();
+
+            if (!employmentProgression.LastModifiedDate.HasValue)
```

Comment:
do this is in SetDefaultValues for consistency once again the model is the only part which updates the model